assets/src/components/v2/disruption_diagram/disruption_diagram.tsx

```diff
@@ -600,7 +599,25 @@ Client is responsible for:
 */
 
 const DisruptionDiagram: ComponentType<DisruptionDiagramData> = (props) => {
-  const { slots, current_station_slot_index, line, effect, svgHeight } = props;
+  const { slots, current_station_slot_index, line, effect } = props;
+
+  // A ref on the diagram container will indicate how much room we have to scale the map
+  const [diagramContainerHeight, setDiagramContainerHeight] = useState(0);
+  const ref = useRef<HTMLDivElement>(null);
+  useEffect(() => {
+    if (!ref.current) return;
+    const resizeObserver = new ResizeObserver(() => {
+      if (ref?.current) {
+        setDiagramContainerHeight(ref.current.clientHeight);
+      }
+    });
+    resizeObserver.observe(ref.current);
+    return () => resizeObserver.disconnect(); // clean up
+  });
```

Comment:
Should this effect have a dependency list? (Maybe `[ref?.current]`?)
Currently, it will run on every re-render, disconnecting the previous observer and connecting a new one.